### ISSUE TITLE
V3 DSL Feedback

### ISSUE THREAD

**Issue by Maxim-Filimonov** [pact-foundation/jest-pact#215]
### Feature description
V3 API.

### Use case
I made the same code using `jest-pact` v3 
```
pactWith(pactConfig, (interaction) => {
  interaction('authentication api', ({ execute, provider }) => {
    const username = mockUsername;
    provider.addInteraction(validUsername({ username }));

    return execute(
      'can check for existence of username',
      async (mockServer) => {
        const response = await validateUsername({
          username,
          api: new Api(mockServer.url),
        });

        expect(response).toBeDefined();
      },
    );
  });
});
```
and `plain pact v3`:
```
describe('authentication', () => {
  it('can check for existence of username', () => {
    const username = mockUsername;
    provider.addInteraction(validUsername({ username }));
    
    return provider.executeTest(async (mockServer) => {
    const response =  await validateUsername({
        username,
        api: new Api(mockServer.url),
     expect(response).toBeDefined();
    });
  });
});
```
It seems to me that plain pact v3 syntax is way closer to jest and easier to understand. I have a clear separation between different interactions and only need to wrap my api calls with special `executeTest` method. 
While with `jest-pact` I'm getting confused as I would need to add NEW interaction for every single interaction that I do even so my interaction already has a name defined. 

I would suggest to remove second layer of interaction and provide `({ execute, provider})` directly in callback of pactWith. I presume `pactWith` just maps to describe, we should be able to use standard `it` blocks to separate tests instead of wrapping them in special syntax.

**Comment by YOU54F** [comment]
Thanks for this @Maxim-Filimonov - I'll take the time to digest soon, but in the meantime happy to hear everyone else's viewpoints on how the new interface is working for them.

Especially considering some of the changes from the Ruby core -> Rust core and the lifecycle changes.

I agree that the experience with using Jest and Pact JS is pretty nice as it is.

>t seems to me that plain pact v3 syntax is way closer to jest and easier to understand. I have a clear separation between different interactions and only need to wrap my api calls with special executeTest method.

**Comment by TimothyJones** [comment]
> It seems to me that plain pact v3 syntax is way closer to jest and easier to understand. 

I agree. I think at the time jest-pact's V3 support was written, the "plain" v3 syntax didn't exist, and it looks like it was jest-pact's V3 support was migrated from beta without reviewing it.

**Comment by TimothyJones** [comment]
By the way, the intention for the interaction block being separate is that you should be able to do this:

```
pactWith(pactConfig, (interaction) => {
  interaction('authentication api', ({ execute, provider }) => {
    beforeEach(() => {
       return provider.addInteraction(validUsername({ username }));
    })
    
   execute(
       /* some positive case */
    );
   execute(
       /* some negative case */
    );
  });
});
```

That could be clearer in the docs - although I'm not actually sure this is that nice. I didn't like the name `execute` at the time.

> While with jest-pact I'm getting confused as I would need to add a NEW interaction for every single interaction that I do even, so my interaction already has a name defined.

Doesn't this happen with your example too? Repeated tests would need the same interaction defined.

> By interaction already has a name I mean my interaction already has:

Yeah. Ideally Jest-Pact should pass this on so you don't have to.

**Comment by johhnsmmith198** [comment]
Hi @Maxim-Filimonov 
You had to define a provider outside your describe section in your `plain pact v3` example.
Did you define the provider in a separated file and have all your test files importing the same provider object? 
Or is it defined in a beforeAll on each test file? 

Why does `jest-pact v3` is deconstructing: `{ execute, provider }` instead of letting us use `provider.executeTest` ? 
Thanks

**Comment by Maxim-Filimonov** [comment]
> Hi @Maxim-Filimonov You had to define a provider outside your describe section in your `plain pact v3` example. Did you define the provider in a separated file and have all your test files importing the same provider object? Or is it defined in a beforeAll on each test file?
> 
> Why does `jest-pact v3` is deconstructing: `{ execute, provider }` instead of letting us use `provider.executeTest` ? Thanks

We just do this before describe block:
```
# pactConfig is just a static object from separate file
const provider = new PactV3(pactConfig);
```
